measure_optimize.py

```diff
@@ -2,180 +2,2412 @@
+
+        # Round possible float solution to be integer
+        for i in range(self.num_measure_dynamic_flatten):
+            for j in range(self.num_measure_dynamic_flatten):
+                if init_cov_y[i][j] > 0.99:
+                    init_cov_y[i][j] = int(1)
+                else:
+                    init_cov_y[i][j] = int(0)
+
+        # initialize m.cov_y with the intial solution
+        # loop over number of measurements
+        for a in range(self.num_measure_dynamic_flatten):
+            # cov_y only have the upper triangle part since solution matrix is symmetric
+            for b in range(a, self.num_measure_dynamic_flatten):
+                self.mod.cov_y[a, b] = init_cov_y[a][b]
+
+        # initialize total manual number. This is not needed since it is initialized by warmstart function
+        # kept for now
+        # total_manual_init = 0
+
+        # initialize the DCM installation flags
+        # this needs initialized since it is binary decisions which warmstart function won't initialize
+        dynamic_install_init = [0, 0, 0]
+
+        # round solutions
+        # if floating solutions, if value > 0.01, we count it as an integer decision that is 1 or positive
+        for i in range(self.n_static_measurements, self.num_measure_dynamic_flatten):
+            if init_cov_y[i][i] > 0.01:
+                # total_manual_init += 1  # kept this line for now
+
+                # identify which DCM this timepoint belongs to, turn the installation flag to be positive
+                i_pos = int((i - self.n_static_measurements) / self.sens_info.Nt)
+                dynamic_install_init[i_pos] = 1
+
+        # initialize m.if_install_dynamic with the value calculated
+        # loop over DCM index
+        for i in self.mod.dim_dynamic:
+            self.mod.if_install_dynamic[i] = dynamic_install_init[
+                i - self.n_static_measurements
+            ]
+
+    def customized_warmstart(self, grey_box=True):
+        """
+        This is the warmstart function provided to MindtPy
+        It is called every mindtpy iteration, after solving MILP master problem, before solving the fixed NLP problem
+        This function initializes all continuous variables in the model given the integer decision values
+
+        Return
+        -----
+        None. Initialize all continuous variables of this model in-place.
+        """
+        # new_fim is the FIM computed with the given integer solution
+        # initialize new_fim
+        new_fim = np.zeros((self.sens_info.n_parameters, self.sens_info.n_parameters))
+
+        if self.optimize_print_level >= 3:
+            # print to see how many non-zero solutions are in the solution
+            for a in self.mod.n_responses:
+                for b in self.mod.n_responses:
+                    if b >= a:
+                        # all solutions that are bigger than 0.001 are treated as non-zero
+                        if self.mod.cov_y[a, b].value > 0.001:
```

Comment:
All parameters like this should be declared and set in the __init__ method for the object

Reply:
set this up in init.